lib/IRGen/IRGen.cpp

```diff
@@ -484,21 +484,17 @@ performOptimizationsUsingNewPassManger(const IRGenOptions &Opts,
   if (Opts.Sanitizers & SanitizerKind::Address) {
     PB.registerOptimizerLastEPCallback([&](ModulePassManager &MPM,
                                            OptimizationLevel Level) {
-      auto Recover = bool(Opts.SanitizersWithRecoveryInstrumentation &
+      AddressSanitizerOptions Opts;
+      Opts.CompileKernel = false;
+      Opts.Recover = bool(Opts.SanitizersWithRecoveryInstrumentation &
                           SanitizerKind::Address);
-      bool UseAfterScope = false;
-      bool ModuleUseAfterScope = true;
-      bool UseOdrIndicator = Opts.SanitizeAddressUseODRIndicator;
-      llvm::AsanDtorKind DestructorKind = llvm::AsanDtorKind::Global;
-      llvm::AsanDetectStackUseAfterReturnMode UseAfterReturn =
-          llvm::AsanDetectStackUseAfterReturnMode::Runtime;
+      Opts.UseAfterScope = true;
```

Comment:
I think `Opts.UseAfterScope` should be set to `false` to mirror what clang defaults to. It seems to default to false unless you add a flag AFAICT.